pinot-spi/src/main/java/org/apache/pinot/spi/ingestion/segment/writer/SegmentWriter.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.ingestion.segment.writer;
+
+import java.io.Closeable;
+import java.io.IOException;
+import org.apache.pinot.spi.data.readers.GenericRow;
+
+
+/**
+ * An interface to collect records and create a Pinot segment.
+ * This interface helps abstract out details regarding segment generation from the caller.
+ */
+public interface SegmentWriter extends Closeable {
+
+  /**
+   * Initializes the {@link SegmentWriter} with provided tableConfig and Pinot schema.
+   */
+  void init(SegmentWriterConfig segmentWriterConfig)
+      throws Exception;
+
+  /**
+   * Collects a single {@link GenericRow} into a buffer.
+   * This row is not available in the segment until a <code>flush()</code> is invoked.
+   */
+  void collect(GenericRow row)
+      throws IOException;
```

Comment:
why would this throw IOException?

Reply:
Changed to Exception. I think IOException came in there from the impl class.